src/libs/react-utils.ts

```diff
@@ -84,7 +84,17 @@ export const useCancellation = (): AbortSignal => {
   return controller.current.signal
 }
 
-export const withDisplayName = _.curry((name, WrappedComponent) => {
+type ComponentWithDisplayName = {
+  (props: any, context?: any): ReactElement<any, any> | null
+  displayName?: string | undefined
+}
+
+type WithDisplayNameFn = {
```

Comment:
The type parameter seems to get lost when going through `safeCurry`, so had to add a separate type for the curried functions.

Reply:
yeah, a two-stage type/fn strategy is commonly required to have good type flow with curry mechanics.